internal/testutil/status/condition.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
```

Comment:
Might it make sense to call this package `statustestutil` to make it clear that it is a test utility? Because these functions don't need a `*testing.T` argument, they kinda look like production code.

Reply:
Actually, in the spirit of package names not being repeated in function names, maybe call it `conditionstestutil`. Then call the functions `SortByType` and `Replace`? So a caller would look like, for example, `conditionstestutil.SortByType()`.

Reply:
I was thinking since `testutil` is already in the path its not ideal to use `util` or `test` again in the file name.
But with imports, since it gets used as `package.Foo` and package won't have `test` or `util` then i may agree with you.  Optimizing for readability when used seems better.

Reply:
```
conditionstestutil.Replace( ... )
conditionstestutil.SortByType( ... )
```